app/api/resources/users.rb

```diff
@@ -1,9 +1,22 @@
 module Resources
   class Users < Grape::API
-    resource :users do
-      desc 'Get all Users'
+    resource :user do
+      desc 'Get current user'
       get do
-        present User.all, with: Entities::User
+        present current_user, with: Entities::User
+      end
+
+      desc 'Validates token <-> user integrity'
+      params do
+        requires :id, type: String, desc: 'id of the user'
+        requires :email, type: String, desc: 'users email'
+      end
+      post :validate do
+        valid_id = current_user.id == params[:id];
+        valid_mail = current_user.email == params[:email]
```

Comment:
Is this also _the place_ where we should check whether or not the token has been expired?

Reply:
no this is done each time a request is performed, [here](https://github.com/lebalz/db-sql/blob/b9420ad9d857be613432a33f6412d4e3f2f8de1e/app/api/api.rb#L41)
I'm not sure if `validate` is really needed or if it is kind of a unnecessary check - it basically should ensure that a login token alone can not be used, but that you do need to know the user id and email too. But since you can request the current user, related to the token, over `api/users`, this check may be nonsens... what you think, could this improve the integrity or could this check be removed?

Reply:
I don't think we need such a check at the moment. So I vote for removing it. Are you okay with that?